blocks/contact-form/contact-form.js

```diff
@@ -0,0 +1,382 @@
+    // the workaround is to create the new script tag, copy attibutes and content
+    const newScript = document.createElement('script');
+
+    newScript.setAttribute('type', 'text/javascript');
+    // coping all script attribute to the new one
+    script.getAttributeNames().forEach((attrName) => {
+      const attrValue = script.getAttribute(attrName);
+      newScript.setAttribute(attrName, attrValue);
+
+      if (attrName === 'src') {
+        waitForLoad = new Promise((resolve) => {
+          newScript.addEventListener('load', resolve);
+        });
+      }
+    });
+    newScript.innerHTML = script.innerHTML;
+    script.remove();
+    document.body.append(newScript);
+
+    // eslint-disable-next-line no-await-in-loop
+    await waitForLoad;
+  }
+
+  const inputs = block.querySelectorAll('input');
+  inputs.forEach((formEl) => {
+    formEl.placeholder = i18n(formEl.placeholder);
+    formEl.ariaLabel = i18n(formEl.ariaLabel);
+  });
+
+  const taEl = block.querySelector('textarea');
+  if (taEl && taEl.placeholder) taEl.placeholder = i18n(taEl.placeholder);
+
+  block.style.display = displayValue;
+
+  const cancelBtn = block.querySelector('.contact-form.block .cta button.cancel');
+  if (cancelBtn) {
+    cancelBtn.addEventListener('click', (e) => {
+      e.preventDefault();
+      e.stopPropagation();
+      hideSideModal();
+    });
+  }
+
+  [...block.querySelectorAll('input[name="first_name"], input[name="last_name"]')]
+    .forEach((el) => {
+      el.addEventListener('blur', (e) => {
+        const { value } = e.currentTarget;
+        if (!value || value.trim().length === 0) {
+          e.currentTarget.classList.add('error');
+        } else {
+          e.currentTarget.classList.remove('error');
+        }
+      });
+    });
+
+  [...block.querySelectorAll('input[name="phone"]')]
+    .forEach((el) => {
+      el.addEventListener('blur', (e) => {
+        const { value } = e.currentTarget;
+        if (!value || value.trim().length === 0 || !phoneRegex.test(value)) {
+          e.currentTarget.classList.add('error');
+        } else {
+          e.currentTarget.classList.remove('error');
+        }
+      });
+      // create input mask
+      el.addEventListener('input', (e) => {
+        const x = e.target.value.replace(/\D/g, '').match(/(\d{0,3})(\d{0,3})(\d{0,4})/);
+        e.target.value = !x[2] ? x[1] : `${x[1]}-${x[2]}${x[3] ? `-${x[3]}` : ''}`;
+      });
+    });
+
+  [...block.querySelectorAll('input[name="email"]')]
+    .forEach((el) => {
+      el.addEventListener('blur', (e) => {
+        const { value } = e.currentTarget;
+        if (!value || value.trim().length === 0 || !emailRegex.test(value)) {
+          e.currentTarget.classList.add('error');
+        } else {
+          e.currentTarget.classList.remove('error');
+        }
+      });
+    });
+};
+
+export default async function decorate(block) {
+  const observer = new IntersectionObserver((entries) => {
```

Comment:
Do we have any pages that use the side load for testing the performance? I want to see how those are handled with this.

Reply:
Added an example in PR description.